app/controllers/purchases_controller.rb

```diff
@@ -78,9 +78,14 @@ def update
 
   def destroy
     purchase = current_organization.purchases.find(params[:id])
-    PurchaseDestroyService.call(purchase)
+    begin
+      PurchaseDestroyService.call(purchase)
+    rescue InventoryError => e
```

Comment:
Is there a reason you can't catch any `StandardError` here?

Reply:
No. I will change it.